src/openrct2/CmdlineSprite.cpp

```diff
@@ -91,16 +91,15 @@ static bool sprite_file_open(const utf8* path)
     if (spriteFileHeader.num_entries > 0)
     {
         int32_t openEntryTableSize = spriteFileHeader.num_entries * sizeof(rct_g1_element_32bit);
-        rct_g1_element_32bit* openElements = static_cast<rct_g1_element_32bit*>(malloc(openEntryTableSize));
+        std::unique_ptr<rct_g1_element_32bit[]> openElements = std::make_unique<rct_g1_element_32bit[]>(openEntryTableSize);
```

Comment:
This seems incorrect, it now has num_entries * sizeof(rct_g1_element_32bit) array elements.

Reply:
Argh! Can you do a quick pr to fix I'm afk

Reply:
I can fix it, but I'm a little confused on how many entries it's supposed to have

Reply:
Just get rid of the sizeof multiplication then its fine.

Reply:
So just to be clear, change line 93 to:
int32_t openEntryTableSize = spriteFileHeader.num_entries sizeof(rct_g1_element_32bit);
and do the same for saveEntryTableSize?

Reply:
malloc is allocating N bytes of memory where as std::make_unique<type[]> allocates N entries of 'type'.

Reply:
Got it. In that case, openEntryTableSize should just equal spriteFileHeader.num_entries, correct?

Reply:
Yeah, might as well just ditch the variable all along.

Reply:
Okay: I committed the changes, but they don't seem to be showing up on GitHub

Reply:
Needs to be on a new branch and PR

Reply:
Thanks! I just pushed the changes. I hope this fixes it!

Reply:
Could you create a PR for this if it's still an issue?

Reply:
Isn't it https://github.com/OpenRCT2/OpenRCT2/pull/12293?

Reply:
@tupaschoal For some reason, most of the messages in this thread were not visible in the tab where I had this open. Must have been some caching issue.

#12293 fixes this, indeed.